beacon-chain/sync/receive_block.go

```diff
@@ -54,26 +56,53 @@ func (rs *RegularSync) receiveBlock(msg p2p.Message) error {
 	defer span.End()
 	recBlock.Inc()
 
-	response := msg.Data.(*pb.BeaconBlockResponse)
+	isValid, blockRoot, err := rs.validateAndProcessBlock(ctx, msg)
+	if err != nil {
+		return err
+	}
+
+	if !isValid {
+		return nil
+	}
+
+	// If the block has a child, we then clear it from the blocks pending processing
+	// and call receiveBlock recursively. The recursive function call will stop once
+	// the block we process no longer has children.
+	if child, ok := rs.hasChild(blockRoot); ok {
+		// We clear the block root from the pending processing map.
+		rs.clearPendingBlock(blockRoot)
+		return rs.receiveBlock(child)
+	}
+	return nil
+}
+
+func (rs *RegularSync) validateAndProcessBlock(ctx context.Context, blockMsg p2p.Message) (bool, [32]byte, error) {
```

Comment:
The thing you want should the the first return.

Please change to [32]byte, bool, error

Reply:
alright, done